plugins/modules/ndo_mac_sec_policy.py

```diff
@@ -0,0 +1,423 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2024, Anvitha Jain (@anvjain) <anvjain@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+DOCUMENTATION = r"""
+---
+module: ndo_mac_sec_policy
+short_description: Manage MACSec Policies on Cisco Nexus Dashboard Orchestrator (NDO).
+description:
+- Manage MACSec Policies on Cisco Nexus Dashboard Orchestrator (NDO).
+- This module is only supported on ND v3.1 (NDO v4.3) and later.
+author:
+- Anvitha Jain (@anvjain)
+options:
+  template:
+    description:
+    - The name of the template.
+    - The template must be a fabric policy template.
+    type: str
+    required: true
+  mac_sec_policy:
+    description:
+    - The name of the MACSec Policy.
+    type: str
+    aliases: [ name ]
+  mac_sec_policy_uuid:
+    description:
+    - The uuid of the MACSec Policy.
+    - This parameter is required when the O(mac_sec_policy) needs to be updated.
+    type: str
+    aliases: [ uuid ]
+  description:
+    description:
+    - The description of the MACSec Policy.
+    type: str
+  admin_state:
+    description:
+    - The administrative state of the MACSec Policy. (Enables or disables the policy)
+    - The default value is C(enabled).
+    type: str
+    choices: [ enabled, disabled ]
+  interface_type:
+    description:
+    - The type of the interfaces this policy will be applied to.
+    type: str
+    choices: [ fabric, access ]
+    default: fabric
+  cipher_suite:
+    description:
+    - The cipher suite to be used for encryption.
+    - The default value is C(256_gcm_aes_xpn).
+    type: str
+    choices: [ 128_gcm_aes, 128_gcm_aes_xpn, 256_gcm_aes, 256_gcm_aes_xpn ]
+  window_size:
+    description:
+    - The window size defines the maximum number of frames that can be received out of order
+    - before a replay attack is detected.
+    - The value must be between 0 and 4294967295.
+    - The default value is 0 for type C(fabric) and 64 for type C(access).
+    type: int
+  security_policy:
+    description:
+    - The security policy to allow trafic on the link for the MACSec Policy.
+    - The default value is C(should_secure).
+    type: str
+    choices: [ should_secure, must_secure ]
+  sak_expiry_time:
+    description:
+    - The expiry time for the Security Association Key (SAK) for the MACSec Policy.
+    - The value must be 0 or between 60 and 2592000.
+    - The default value is 0.
+    type: int
+  confidentiality_offset:
```

Comment:
This one seems to only apply for fabric interface_type

Reply:
changed